### DISCUSSION TITLE
Extract coordinates of central sulcus

### DISCUSSION THREAD

**Discussion by MultimodalBrainMonitoringUDA** [brainvisa/brainvisa.github.io#48] [Q&A] [answered]
I would like to export the coordinates of the central sulcul (after the automated procedure for the recognition). Is there a way to do this?
Or can someone suggest me how to do?
In alternative also saving the sulcus as a mesh could works for me.

**Comment by MultimodalBrainMonitoringUDA** [comment]
Specifically I would like to have the coordinates of the central sulcus in the mesh space

**Comment by denisri** [comment]
Hi,

It's not clear which coordinates you have in mind, we can think about lots of different coordinates: an average point (mass center), extremities, coordinates of each point etc. and also in which coordinates system you want them: native image coordinates, inter-subject space (Talairach or MNI...) etc.
We use sulci as a structured objects set (a graph) which contains vertices, a vertex represents a fold part (a sulcus part). Thus a sulcus is made up of several vertices with the same label (the label of the sulcus). A vertex can store lists of voxels (what we call "buckets") and meshes, in native image space. The graph itself contains coordinates transformation information on how to get to Talairach space. So all the information you are looking for is potentially present in this graph. In BrainVIsa morphologist toolboox there is a morphometry process which can provide some of this information, but if you need more precise things, it probably requires a bit of scripting / programming. The BrainVisa distribution comes with Python libraries which allow to manipulate the graphs, buckets and mesh objects. Have a look here for instance: https://brainvisa.info/pyaims/sphinx/pyaims_tutorial_nb.html#Graphs

Denis

**Reply by MultimodalBrainMonitoringUDA** [reply]
Hi, thanks for the reply.
I would like to have the coordinates of each point of the Central Sulcus in the native image coordinates. Could you provide me some script snippets to do this? a sort of starting point. I'm vary new to BrainVIsa. Thank again,
Pierpaolo

**Comment by denisri** [comment, marked as answer]
for instance:
```
from soma import aims
import numpy as np

graph = aims.read('filename.arg')
sulci_names = set(['S.C._left', 'S.C._right'])  # extend if needed
sum_coords = np.zeros((3, ))
npts = 0
for v in graph.vertices():
    if v.get('label') in sulci_names:
        mesh = v.get('aims_Tmtktri')
        if mesh:
            vert = mesh.vertex()
            sum_coords += np.sum(np.asarray(vert), axis=0)
             npts += len(vert)
avg_coords = sum_coords / npts
```
coords are in native space, in the Aims images convention (https://brainvisa.info/aimsdata-5.0/user_doc/tutorial.html#handling-referentials-and-transformations)
